Meeqan/buys
# #162530905 #162530882 User registrations and login

**What does this PR do?**

This PR implements the User authentication endpoints.

**Description of Task to be completed?**
As a developer, I should be able to generate a JWT token by making a `POST` request with the valid details to `/user/register` for registration or `user/login` for login.


**How should this be manually tested?**
- Pull the branch `git fetch origin ft-user-registration-162530882`
- Check into it` git checkout ft-user-registration-162530882`
- Activate the pipenv
- Run the app: python manage.py runserver
- On postman make a POST request to `{{base-url}}/api/v1/user/login` or `{{base-url}}/api/v1/user/register`

body - Register
```
{
    "email": "example@gmail.com",
    "password": "Password@123",
    "first_name": "Test",
    "last_name": "User"
}
```

body - Login
```
{
    "email": "example@gmail.com",
    "password": "Password@123",
}
```

**Any background context you want to provide?**
N/A

**What are the relevant pivotal tracker stories?**
[#162530905](https://www.pivotaltracker.com/n/projects/2170023/stories/162530905)
[#162530882](https://www.pivotaltracker.com/n/projects/2170023/stories/162530882)
